app/javascript/controllers/auto_submit_controller.js

```diff
@@ -0,0 +1,17 @@
+import { Controller } from "@hotwired/stimulus"
+
+// Connects to data-controller="auto-submit"
+export default class extends Controller {
+  static targets = ['form']
+
+  submit() {
+    const form = this.formTarget
+    let timeoutID;
+
+    clearTimeout(timeoutID)
+    timeoutID = setTimeout(() => {
```

Comment:
- これ毎回 timeoutID が作られているから、clearTimeout は機能してない。
  インスタンス変数にしないと、次の submit の時に保持しない。
-  timeoutID が存在しない場合は clear したくない
- timeoutId としたい

```suggestion
    if(this.timeoutId) { 
      clearTimeout(timeoutId)
    }
    this.timeoutId = setTimeout(() => {
```

Reply:
確かにそうです。今のままだと同じtimeoutIdとして認識されていないですね。